build-logic/src/main/kotlin/com/ibm/wala/gradle/JavaCompileUsingEcj.kt

```diff
@@ -65,10 +65,18 @@ abstract class JavaCompileUsingEcj : JavaCompile() {
 
   @TaskAction
   protected override fun compile(inputs: InputChanges) {
+    val testArgs = options.allCompilerArgs
+    val argFile = createTempFile(prefix = "kotlinTemp", suffix = ".tmp")
```

Comment:
`createTempFile` is deprecated. Use `kotlin.io.path.createTempFile` instead.

Reply:
Can we use that call with the same arguments as `createTempFile`.  I thought this was a bad thing to use, but the documentation I found online was not too helpful.

Reply:
Yes, you can use the same arguments. So what's the difference?

* `kotlin.io.createTempFile` is implemented using `java.io.File.createTempFile`, which is sloppy about permissions on the temporary file.
* `kotlin.io.path.createTempFile` is implemented using `java.nio.file.Files.createTempFile`, which is more careful about permissions on the temporary file.